### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by chuneuny-emily**
### Summary
Apply PR template based on https://intel.sharepoint.com/:p:/s/IOTGVMCAVA/EQ9i39-Wt5ZGmHLR3dIbqtABdKlFyWG4Iz3i0g36GjCwNw?e=ore0ld&CID=74933166-1592-4CF6-B118-A8D3630413BE&wdLOR=c01BFE16A-A5C5-4D9E-837A-4F4AF9467723

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -18,11 +19,12 @@ not fully covered by unit tests or manual testing can be complicated. -->
...
-- [ ] I submit my changes into the `develop` branch
-- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
+- [ ] I have added unit tests to cover my changes.​
+- [ ] I have added integration tests to cover my changes.​
+- [ ] I have added e2e tests for validation.
+- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/releases/1.1.0/CHANGELOG.md).​
 - [ ] I have updated the [documentation](
-  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
-- [ ] I have added tests to cover my changes
+  https://github.com/openvinotoolkit/datumaro/tree/releases/1.1.0/site) accordingly
```
Comment (wonjuleee) [inline .github/pull_request_template.md]: here also.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -18,11 +19,12 @@ not fully covered by unit tests or manual testing can be complicated. -->
 
 ### Checklist
 <!-- Put an 'x' in all the boxes that apply -->
-- [ ] I submit my changes into the `develop` branch
-- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
+- [ ] I have added unit tests to cover my changes.​
+- [ ] I have added integration tests to cover my changes.​
+- [ ] I have added e2e tests for validation.
+- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/releases/1.1.0/CHANGELOG.md).​
```
Comment (wonjuleee) [inline .github/pull_request_template.md]: should we update this for every release?

File: .github/pull_request_template.md (hunk 1 above)
Comment (vinnamkim) [reply .github/pull_request_template.md]: We are able to just keep this file in `develop` branch, not `release` branch.

Comment (wonjuleee) [reply .github/pull_request_template.md]: agree.

File: .github/pull_request_template.md (hunk 2 above)
Comment (chuneuny-emily) [reply .github/pull_request_template.md]: > should we update this for every release?
I think so. If RC branch is created, if there are some changes, the default PR 
 template will link to develop's one. I think we should link the proper version's one.  @yunchu What do you think of it? If not, every default template will link to develop branch.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -18,11 +19,12 @@ not fully covered by unit tests or manual testing can be complicated. -->
...
 <!-- Put an 'x' in all the boxes that apply -->
-- [ ] I submit my changes into the `develop` branch
-- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
-- [ ] I have updated the [documentation](
-  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
-- [ ] I have added tests to cover my changes
+- [ ] I have added unit tests to cover my changes.​
+- [ ] I have added integration tests to cover my changes.​
+- [ ] I have added e2e tests for validation.
+- [ ] I have added the description of my changes into CHANGELOG in my target branch. (e.g. [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md) in develop)​
```
Comment (wonjuleee) [inline .github/pull_request_template.md]: please modify `e.g.` to `e.g.,` (https://en.wiktionary.org/wiki/e.g.)

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -18,11 +19,12 @@ not fully covered by unit tests or manual testing can be complicated. -->
...
+- [ ] I have updated the documentation in my target branch accordingly. (e.g. [documentation](
```
Comment (wonjuleee) [inline .github/pull_request_template.md]: here also.

**Review by wonjuleee — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Pull request merged**